[CI] Added CI run on minimum and current python requirements

DjLegolas opened the pull request:
CI will run on both python 3.6 and 3.8

DjLegolas commented:
Seem as there is no way to test on python3.6 because there is no libtorrent binding to 3.6 for windows or linux...

doadin commented:
imo 3.10 has been working nicely, I think we could change this from 3.7-3.10.

DjLegolas commented:
I'm thinking about upgrading the minimum python version, again, to 3.7 and run the CI on all the supported versions (3.7-3.10)...
@cas-- @gazpachoking @aresch what do you guys think?

gazpachoking commented:
I'm 👍 on dropping 3.7 support. Not sure when we'd want to fit that in in terms of versioning and releases. In terms of this PR, I'd say testing more than one version is an improvement, regardless of whether it hits every currently supported version. I'm also 👍 on adding in other versions up to 3.10

doadin commented:
should also maybe add fail-fast: false

DjLegolas commented:
While testing, it helps having everything fail fast.
But will add that flag later, thanks @doadin

doadin commented:
3.10 needs pytest >= 6.2.5 I think.

doadin commented:
https://github.com/deluge-torrent/deluge/blob/develop/requirements-tests.txt#L1
pytest == 6.2.5 ; python_version >= '3.10'
pytest != 5.2.3, < 5.4
maybe

DjLegolas commented:
No, I'll just drop the limits from `pytest`...
It worked on my local Windows and Ubuntu, both with python 3.8.

DjLegolas commented:
To sums this up:
* Testing python versions 3.7-3.10
* Switched `libtorrent`'s installation source to `pypi`
* Dropped `pytest` version limits, current version is `7.0.0` for python 3.10 (thanks @doadin for the heads up)

doadin commented:
I am noticing a few things we use like twisted in a coming version is droping python3.6 and pyinstaller, Pillow already has, could probabbly drop 3.6 our selves as well. if it was only pyinstaller then id say no but deps are starting to drop as well. At least on the windows side requiring 3.7+ only excluded windows xp I think which nobody should still be using so not really a big deal imo.

cas-- commented on .github/workflows/ci.yml:
```diff
@@ -10,6 +10,10 @@ on:
   test-linux:
     runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
```
Let's keep fail-fast on

DjLegolas replied:
Done

cas-- replied:
What I meant was, can we remove this line

cas-- commented on .github/workflows/ci.yml:
```diff
@@ -10,6 +10,10 @@ on:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
can we test just 3.7 and 3.10

cas-- commented on .github/workflows/ci.yml:
```diff
@@ -32,22 +36,14 @@ jobs:
       - name: Install dependencies
         run: |
           pip install --upgrade pip wheel
+          pip install libtorrent==1.2.*
```
let's move this to the requirements.txt and remove the version requirement

DjLegolas replied:
Just a reminder, removing the version requirement will install libtorrent 2

cas-- commented on .github/workflows/ci.yml:
```diff
@@ -68,7 +64,7 @@ jobs:
         run: |
           ulimit -c unlimited  # Enable core dumps to be captured
-          cp /usr/lib/python3/dist-packages/libtorrent*.so $GITHUB_WORKSPACE/deluge
+          cp $LD_LIBRARY_PATH/python3*/site-packages/libtorrent*.so $GITHUB_WORKSPACE/deluge
```
this lib copy should not be needed if using pip install

DjLegolas replied:
Removed.

cas-- reviewed (commented):
A few suggestions to make to both linux and win runs
